mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/impl/MongoDBTypeContributor.java

```diff
@@ -20,5 +20,6 @@
 	@Override
 	public void contribute(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
 		typeContributions.contributeType( new StringAsObjectIdType() );
+		typeContributions.contributeType( new BytesAsBsonBinaryType() );
```

Comment:
I think you can avoid this change and in the dialect refers directly to BinaryType instead.